vendor/sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client/client.go

```diff
@@ -129,7 +139,12 @@ func (t *grpcTunnel) serve(c clientConn) {
 			t.connsLock.RUnlock()
 
 			if ok {
-				conn.readCh <- resp.Data
+				select {
+				case conn.readCh <- resp.Data:
+				case <-time.After((time.Duration)(t.readTimeoutSeconds) * time.Second):
```

Comment:
it looks like this is hit once per data packet, which seems high-frequency. `time.After` does not release resources until the time fires (https://golang.org/pkg/time/#After)

> The underlying Timer is not recovered by the garbage collector until the timer fires. If efficiency is a concern, use NewTimer instead and call Timer.Stop if the timer is no longer needed.

Reply:
/cc @caesarxuchao

Reply:
Resolved with https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/220 (Post v0.0.18)